projects/plugins/jetpack/modules/subscriptions/jetpack-user-content-link-redirection.php

```diff
@@ -9,17 +9,38 @@
  * @package automattic/jetpack
  */
 
+use Automattic\Jetpack\Connection\Client;
+
 /**
  * Render a page containing an iframe to track and redirect the user content link in emails.
  */
 function jetpack_user_content_link_redirection() {
 	if ( empty( $_SERVER['QUERY_STRING'] ) ) {
 		return;
 	}
+
+	// phpcs:ignore WordPress.Security.NonceVerification.Recommended
+	if ( isset( $_GET['user_content_redirect_domain_checking'] ) ) {
+		// phpcs:ignore WordPress.Security.NonceVerification.Recommended
+		if ( empty( $_SERVER['QUERY_STRING'] ) || ! isset( $_SERVER['HTTP_HOST'] ) || ! isset( $_GET['blog_id'] ) ) {
+			wp_safe_redirect( get_home_url() );
+			exit();
+		}
+
+		$domain_from_request = sanitize_text_field( wp_unslash( $_SERVER['HTTP_HOST'] ) );
+		// phpcs:ignore WordPress.Security.NonceVerification.Recommended
+		$blog_id = absint( wp_unslash( $_GET['blog_id'] ) );
+
+		if ( ! jetpack_user_content_link_is_same_domain( $domain_from_request, $blog_id ) ) {
```

Comment:
Is this function already calls the WPCOM server, why not just perform all the checking there and redirect here without the iframe flow? Unless, you're keeping the iframe to ensure we can keep track of other data (browser, device, location, etc.)?

Reply:
I tried this solution, but I didn't succeed to redirect to a external link using wp_safe_redirect. Not sure if there are limitations for that (maybe that is the reason for the iframe flow). I'd prefer not changing much with this PR, just focusing on the fix, and we can iterate on that on a follow-up task. What do you think?

Reply:
>  I'd prefer not changing much with this PR, just focusing on the fix, and we can iterate on that on a follow-up task. What do you think?

IMO, I wished there was a way not to call home everything a link is clicked on.

> I tried this solution, but I didn't succeed to redirect to a external link using wp_safe_redirect.

Oh yes, in this case likely a `wp_redirect` would work over `wp_safe_redirect` as the URL could be anything.

>  (maybe that is the reason for the iframe flow)

The reason for the iframe flow was mainly to have as much logic as possible done on WPCOM end so we don't need to worry about rolling out a fix that can only get in 1-2 weeks time with Jetpack's release timeline. The other reason (unconfirmed) is to track other things like browser, device location and such.